lib/blockscore/actions/update.rb

```diff
@@ -27,20 +32,39 @@ module Update
 
       def_delegators 'self.class', :endpoint, :patch
 
+      # May be deprecated in future releases, prefer :update.
       def save!
         if persisted?
-          patch("#{endpoint}/#{id}", filter_params)
+          patch(update_url, filter_params)
           true
         else
           super
         end
       end
 
+      def update(params)
+        update!(params)
+        true
+      rescue
+        false
+      end
+
+      def update!(params)
+        response = patch(update_url, params)
+        self.attributes = response.attributes
+      end
```

Comment:
I think it is very possible that we could, sure. It could be any of them, but the most likely would be `APIError` or `InvalidRequestError`. I think it really just comes down to how you want the gem to behave: should the gem itself do the error handling in some way, or should it be passed onto the user and allow them to deal with the different errors however they see fit. I think you could make a strong argument either way.

Just let me know what approach you want to take.